algebird-core/src/main/scala/com/twitter/algebird/HyperLogLogSeries.scala

```diff
@@ -31,12 +31,59 @@ package com.twitter.algebird
  * @return  New HLLSeries
  */
 case class HLLSeries(bits: Int, rows: Vector[Map[Int, Long]]) {
+
+  def insert(data: Array[Byte], timestamp: Long): HLLSeries = {
+    import HyperLogLog._
+    val hashed = hash(data)
+    val j0 = j(hashed, bits)
+    val rhow = rhoW(hashed, bits) - 1
+
+    val newRows = if (rhow < rows.size) {
+      val row = rows(rhow)
+      val t = row.get(j0) match {
+        case Some(t0) => Math.max(t0, timestamp)
+        case None => timestamp
+      }
+      rows.updated(rhow, row.updated(j0, t))
+    } else {
+      val padded = (rows.size until rhow).foldLeft(rows)((rs, _) => rs :+ Map.empty)
+      padded :+ Map(j0 -> timestamp)
+    }
+    HLLSeries(bits, newRows)
+  }
+
+  def maxRhowStats(threshold: Long): (Int, Double) = {
+    val seen = scala.collection.mutable.Set.empty[Int]
+    var sum: Double = 0.0
+    var i = rows.size - 1
+    while (i >= 0) {
+      val it = rows(i).iterator
+      while (it.hasNext) {
+        val (k, t) = it.next
+        if (t >= threshold && !seen(k)) {
```

Comment:
Ah, of course, for some reason I missed that we were adding it into sum :| That said, you can still skip this check by using `add`, which returns `false` if the set already contains the element:

```scala
if (t >= threshold && seen.add(k)) {
  sum += HyperLogLog.negativePowersOfTwo(i + 1)
}
```